File: 2018_prelim/tasks/island/solutions/ranald/st4_ranald.cpp
#include <bits/stdc++.h>
using namespace std;
const int MAXN = 200004;
int N, M, cnt[MAXN], fac[MAXN];
int main() {
	scanf("%d%d", &N, &M);
	for (int i = 0, u, v; i < N+M-1; ++i) {
		scanf("%d%d", &u, &v);
		cnt[u-1]++;
		cnt[v-1]++;
	}
	for (int i = 0; i < N+M; ++i) {
		fac[cnt[i]-1]++;
	}
	for (int i = N+M-1; i >= 1; --i) {
		if (fac[i] > 0) {
			printf("%d %d\n", i, fac[i]);
			fac[i-1] += fac[i];
		}
		
	}

}

File: 2018_prelim/tasks/island/solutions/ranald/st4_ranald.py
N, M = map(int, raw_input().split())
cnt = [0]*(N+M+1)
for i in range(N+M-1):
	u, v = map(int, raw_input().split())
	cnt[u] = cnt[u] + 1
	cnt[v] = cnt[v] + 1
	
fac = [0]*(N+M+1)
for i in range(1, N+M+1):
	fac[cnt[i]-1] += 1

for k in range(N+M, 1, -1):
	if fac[k] > 0:
		fac[k-1] += fac[k]
		print ("%d %d" % (k, fac[k]))
